aspnetcore/fundamentals/servers/kestrel.md

```diff
@@ -997,6 +1047,48 @@ The following examples use the <xref:Microsoft.AspNetCore.Server.Kestrel.Core> n
 using Microsoft.AspNetCore.Server.Kestrel.Core;
 ```
 
+Kestrel options, which are configured in C# code in the following examples, can also be set using a [configuration provider](xref:fundamentals/configuration/index). For example, the File Configuration Provider can load Kestrel configuration from a file (*appsettings.json* or *appsettings.{Environment}.json*):
+
+```json
+{
+  "Kestrel": {
+    "Limits": {
+      "MaxConcurrentConnections": 100,
+      "MaxConcurrentUpgradedConnections": 100
+    }
+  }
+}
+```
+
+Use **one** of the following approaches:
+
+* Inject an instance of `IConfiguration` into the `Startup` class from the `Microsoft.Extensions.Configuration` namespace (`Configuration` in the following example). In `Startup.ConfigureServices`, load the `Kestrel` section of configuration into Kestrel's configuration:
```

Comment:
2 bullets

show all the code

* Inject an instance of `IConfiguration` into the `Startup` class.
~from the `Microsoft.Extensions.Configuration` namespace (`Configuration` in the following example).~
* In `Startup.ConfigureServices`, load the `Kestrel` section of configuration into Kestrel's configuration: